### DISCUSSION TITLE
metamask bug: Failing contract deployments

### DISCUSSION THREAD

**Discussion by mpeyfuss** [wevm/viem#1605] [General]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Package Version

1.20.3

### Current Behavior

When using MetaMask version 11.7.0, contract deployments fail due to intrinsic gas being set too low. This doesn't happen on version 11.6.2 or other wallets like Rabby and Rainbow. This seems to be a MM issue and I've logged a bug there, but wanted to bring it up here as well so Viem developers are aware.

### Expected Behavior

Can deploy a contract

### Steps To Reproduce

1. Go to https://stackblitz.com/github/wevm/viem/tree/main/examples/contracts_deploying-contracts
2. Wait for the example to build
3. Connect MM
4. Click deploy
5. Submit tx
6. See that tx failed in MM
7. Open console to inspect the failure

### Link to Minimal Reproducible Example (StackBlitz, CodeSandbox, GitHub repo etc.)

https://stackblitz.com/github/wevm/viem/tree/main/examples/contracts_deploying-contracts

### Anything else?

MM issue: https://github.com/MetaMask/metamask-extension/issues/22359

**Comment by realhen** [comment]
Hi, I am also facing this issue. I am not able to deploy a contract due to the intrinsic gas being set too low. There is the "gas" property on `deployContractO()` that you are able to set which solves this issue, but I don't think there is a way to estimate gas of a contract deployment in Viem. Setting the "gas" property will change the gasLimit, but it doesn't tell the user that its a site suggested limit (below is an example of what I mean when a site suggests gas).

![image](https://github.com/wevm/viem/assets/41842805/4f57da82-4a5e-4e6e-ab8f-f178b143b1dd)


The only methods seem to be https://viem.sh/docs/contract/estimateContractGas.html and https://viem.sh/docs/actions/public/estimateGas.html but these are for already deployed contracts.
